### PR TITLE
Convert controller statefulset files to structured logs

### PR DESCRIPTION
**Pull request by scottilee**
#### What type of PR is this?
/kind cleanup
/sig instrumentation

#### What this PR does / why we need it:
Converts log statements to use structured logging.

#### Which issue(s) this PR fixes:
N/A

#### Special notes for your reviewer:
N/A

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:
```docs
- [KEP]: https://github.com/kubernetes/enhancements/blob/master/keps/sig-instrumentation/1602-structured-logging/kep.yaml
```

### CODE REVIEW

File: pkg/controller/statefulset/stateful_set.go [hunk 1]
```diff
@@ -143,8 +143,8 @@ func (ssc *StatefulSetController) Run(workers int, stopCh <-chan struct{}) {
 	defer utilruntime.HandleCrash()
 	defer ssc.queue.ShutDown()
 
-	klog.Infof("Starting stateful set controller")
-	defer klog.Infof("Shutting down statefulset controller")
+	klog.Infof("Starting StatefulSet controller")
```
Comment (yangjunmyfm192085) [inline pkg/controller/statefulset/stateful_set.go:146]:
```suggestion
	klog.InfoS("Starting StatefulSet controller")
```

Comment (scottilee) [reply pkg/controller/statefulset/stateful_set.go:146]: Fixed.

**Comment by scottilee** [comment]
@yangjunmyfm192085 Thanks for the review! Fixed.

File: pkg/controller/statefulset/stateful_set.go [hunk 2]
```diff
@@ -219,7 +219,7 @@ func (ssc *StatefulSetController) updatePod(old, cur interface{}) {
 		if set == nil {
 			return
 		}
-		klog.V(4).Infof("Pod %s updated, objectMeta %+v -> %+v.", curPod.Name, oldPod.ObjectMeta, curPod.ObjectMeta)
+		klog.V(4).InfoS("Pod updated.", "curPod", klog.KObj(curPod), "prevPod", klog.KObj(oldPod))
```
Comment (yangjunmyfm192085) [inline pkg/controller/statefulset/stateful_set.go:222]:
```suggestion
		klog.V(4).InfoS("Pod updated", "curPod", klog.KObj(curPod), "prevPod", klog.KObj(oldPod))
```

File: pkg/controller/statefulset/stateful_set.go [hunk 3]
```diff
@@ -267,7 +267,7 @@ func (ssc *StatefulSetController) deletePod(obj interface{}) {
 	if set == nil {
 		return
 	}
-	klog.V(4).Infof("Pod %s/%s deleted through %v.", pod.Namespace, pod.Name, utilruntime.GetCaller())
+	klog.V(4).InfoS("Pod deleted through caller.", "pod", klog.KObj(pod), "caller", utilruntime.GetCaller())
```
Comment (yangjunmyfm192085) [inline pkg/controller/statefulset/stateful_set.go:270]:
```suggestion
	klog.V(4).InfoS("Pod deleted through caller", "pod", klog.KObj(pod), "caller", utilruntime.GetCaller())
```

**Comment by yangjunmyfm192085** [comment]
A little change, then lgtm

File: pkg/controller/statefulset/stateful_set.go (hunk 2 above)
Comment (scottilee) [reply pkg/controller/statefulset/stateful_set.go:222]: Fixed.

File: pkg/controller/statefulset/stateful_set.go (hunk 3 above)
Comment (scottilee) [reply pkg/controller/statefulset/stateful_set.go:270]: Fixed.

**Comment by scottilee** [comment]
@yangjunmyfm192085 Fixed.

**Comment by scottilee** [comment]
Rebased.

**Pull request closed without merging**